app/routes.py

```diff
@@ -1 +1,219 @@
+        "token": os.environ.get("SLACK_BOT_TOKEN")
+    }
+
+    r = requests.post(url='https://slack.com/api/chat.postMessage',data=kim_bot)
+
+    return {"task": task.to_result()}, 200
+
+# Mark Task as Incompleted Route
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def update_task_as_incomplete(task_id):
+    task = validate_item(Task, task_id)
+
+    request_data = request.get_json()
+    task.completed_at = None
+
+    db.session.commit()
+
+    return {"task": task.to_result()}, 200
+
+
+# *****************************************************************
+# **********************Goal Routes********************************
+# *****************************************************************
+
+# Create Goal Route
+@goal_bp.route("", methods=["POST"])
+def add_goal(): 
+    response_body = request.get_json()
+
+    # TODO - from_dict method
+    try:
+        new_goal = Goal(title=response_body["title"])
+    except KeyError: 
+        return {"details": "Invalid data"}, 400
+    
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+# Get Goal Route
+@goal_bp.route("", methods=["GET"])
+def get_goals(): 
+    response = [] 
+    all_goals = Goal.query.all()
+
+    for goal in all_goals: 
+        response.append(goal.to_dict())
+    
+    return jsonify(response), 200
+
+# Get One Goal Route
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id): 
+    goal = validate_item(Goal, goal_id)
+
+    return {"goal": goal.to_dict()}, 200
+
+# Update a Goal Route
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+    request_data = request.get_json()
+    goal.title = request_data["title"]
+
+    db.session.commit()
+
+    return {"goal": goal.to_dict()}, 200
+
+# Delete a Goal Route 
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return {"details": f"Goal {goal_id} \"{goal.title}\" successfully deleted"}, 200
+
+
+# Adding Tasks to a Goal
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def add_tasks_to_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+    request_body = request.get_json()
+    all_tasks = request_body["task_ids"]
+
+    for task_id in all_tasks: 
+        task = validate_item(Task, task_id)
+        task.goal = goal
+    
+    db.session.commit()
+
+    return jsonify({"id": goal.goal_id, 
+                    "task_ids": all_tasks}), 200
+
+# Get Tasks of One Goal
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_all_tasks_of_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+    tasks = []
+
+    for task in goal.tasks:
+        tasks.append(task.to_result())
+    
+    return jsonify({"id": goal.goal_id,
+            "title": goal.title, 
+            "tasks": tasks}), 200
```

Comment:
Small note here, but you can still use the `goal.to_dict()` here. You just need to add the list of tasks to the dictionary before returning it.